server/game/cards/events/02/trialbycombat.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../../drawcard.js');
+const ApplyClaim = require('../../../gamesteps/challenge/applyclaim.js');
+
+class TrialByCombat extends DrawCard {
+    setupCardAbilities() {
+        this.interrupt({
+            when: {
+                onClaimApplied: (event, challenge) => (
+                    challenge.winner === this.controller &&
+                    challenge.attackingPlayer === this.controller &&
```

Comment:
The winner / attacking player restrictions aren't technically in the card text, but this is probably a reasonable change from a playability stand point (most of the time you wouldn't want to trigger on yourself, but there are edge cases). We can revisit it if we ever implement Melee or if there are complaints.